org.eclipse.jdt.core/dom/org/eclipse/jdt/internal/core/dom/ICompilationUnitResolver.java

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.jdt.internal.core.dom;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.WorkingCopyOwner;
+import org.eclipse.jdt.core.dom.ASTRequestor;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.FileASTRequestor;
+import org.eclipse.jdt.internal.compiler.batch.FileSystem.Classpath;
+
+/**
+ * This interface is used to resolve a jdt dom tree from source files.
+ * It is contributed to via the compilationUnitResolver extension point.
+ * This interface is currently internal only, and is not considered API.
+ * This interface may be modified, changed, or removed at any time.
+ *
+ * @since 3.38
+ */
+public interface ICompilationUnitResolver {
```

Comment:
This is in internal package, meant as provisional API? doesn't that contradict plugin mechanism?

Reply:
The extension point itself is provisional: https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2560/files#diff-593bbeeaa5bd91d942bb0a52abe6bc4829af2b970e665caa842759bf0ff7691fR9 says "This extension point is not considered API.". We hope to make it API later if there is interest, when we have consolidated with the Javac experience, but while it is already valuable enough to have an extension point, it's still too early to guarantee API stability.

Reply:
then the javadoc has to clearly state so

Reply:
Javadoc already says " This interface is currently internal only, and is not considered API. This interface may be modified, changed, or removed at any time.". What exactly do you think is missing?

Reply:
`<strong>EXPERIMENTAL</strong>.`
https://github.com/eclipse-platform/eclipse.platform/blob/master/docs/Provisional_API_Guidelines.md